File: src/misc/credentials/Credentials.js
// @flow

import type {Base64, Base64Url} from "@tutao/tutanota-utils"

/** Data obtained after logging in. */
export type Credentials = {
	/**
	 * Identifier which we use for logging in.
	 * Email address used to log in for internal users, userId for external users.
	 * */
	login: string,
	/** Session#accessKey encrypted password. Is set when session is persisted. */
	encryptedPassword: ?Base64,
	accessToken: Base64Url,
	userId: Id,
	type: "internal" | "external",
}

File: app-android/tutashared/src/main/java/de/tutao/tutashared/push/Credentials.kt
package de.tutao.tutashared.push

import de.tutao.tutasdk.CredentialType
import de.tutao.tutasdk.Credentials
import de.tutao.tutashared.ipc.UnencryptedCredentials

/**
 * Convert UnencryptedCredentials to credentials for logging in with Sdk.
 */
fun UnencryptedCredentials.toSdkCredentials() = Credentials(
	login = this.credentialInfo.login,
	userId = this.credentialInfo.userId,
	accessToken = this.accessToken,
	encryptedPassphraseKey = this.encryptedPassphraseKey!!.data,
	credentialType = CredentialType.INTERNAL
)